homeassistant/helpers/entity.py

```diff
@@ -607,6 +619,9 @@ def _async_write_ha_state(self) -> None:
         if (icon := (entry and entry.icon) or self.icon) is not None:
             attr[ATTR_ICON] = icon
 
+        if (translation_key := self.translation_key) is not None:
```

Comment:
Shouldn’t we store this in the entity registry instead of the state? It’s not changing.

Reply:
Don't we want this to work also for entities which are not in the entity registry though?

Reply:
If we're going to key this by domain of integration that is providing the entity (as per other comment thread), we will need the entity registry anyway?

I think that it's fine to require a unique ID to be able to use advanced translation features.

Reply:
No, we don't need the entity registry for that, frontend knows which integration an entity belongs to regardless of it's in the entity registry or not.

Reply:
Oh right. But that's an extra data set that needs to be fetched. 

I still think it's weird that if we would put it in the state ? If we were to add device classes today, it would also go into the registry.